services/coincap/coincap-rank.tester.js

```diff
@@ -0,0 +1,13 @@
+import { createServiceTester } from '../tester.js'
+export const t = await createServiceTester()
+
+t.create('rank').get('/bitcoin.json').expectBadge({
+  label: 'Bitcoin',
+  message: '1',
```

Comment:
For the tests that call a live service, we sholdn't hard-code a response in the test. Lets validate that this as `Joi.number().integer().min(1)`

Reply:
The .json return a string even being number, i had to validated it with regex.

Reply:
You can still use `Joi.number()` to picture-check a string.

Try this as a minimal example:

```js
import Joi from 'joi'

const schema = Joi.number().integer().min(1).required()

Joi.attempt('1', schema); // passes
Joi.attempt('0', schema); // fails
```

Same goes for the changepercent24hr and priceusd schemas - they can use `Joi.number()` instead of a regex which should make things a bit neater.

Reply:
Nice! but to changepercent24hr and priceusd `Joi.number().required()` doesn't worked.